doc/cpp/static-grpc-object.md

```diff
@@ -0,0 +1,38 @@
+# gRPC Objects with static storage duration
+
+In general, it's discouraged to define gRPC objects with static storage
+duration because it can have a indeterministic crash due to certain
+initialization order or termination order.
+This is a well known issue as describied in
+[Static Initialization Order Fiasco](https://en.cppreference.com/w/cpp/language/siof)
+and guided in
+[Static and Global Variables](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables)
+
+## Example
+
+The following code may crash because `cq`, which is a `grpc::CompleitionQueue`
+type is defined as a global static variable which is going to be initialized
+before `main()`.
+To initialize `grpc::CompletionQueue` type object, it needs fully initialized
+gRPC system but it may or may not happen at that point.
+In addition to that, when `cq` is destroyed, it also needs alive gRPC system
+which may be terminiated already.
+
+```
+static grpc::CompletionQueue cq;
```

Comment:
Can I write:
```
static std::unique_ptr<grpc::CompletionQueue> cq;

int main() {
  cq.reset(new grpc::CompletionQueue());
  cq->Shutdown();
}
```

Reply:
This code is for a bad example (cq created before main()) but cq in your code will be created in main(). Do you show it as a solution?

Reply:
But it will be destroyed after main... I posit this mainly to figure out the bounds of what we'd support.
For example, I've certainly encouraged folks to have a singleton channel client side to do most of their requests - as it tends to avoid the anti-pattern of creating a new channel per request (which crops up frequently!).

Reply:
Oh your code here is for a solution, right?